File: qc-domain/src/main/java/com/eager/questioncloud/domain/review/dto/MyQuestionReview.java
package com.eager.questioncloud.domain.review.dto;

import com.eager.questioncloud.domain.review.model.QuestionReview;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class MyQuestionReview {
    private Long id;
    private Integer rate;
    private String comment;

    public static MyQuestionReview from(QuestionReview questionReview) {
        return new MyQuestionReview(questionReview.getId(), questionReview.getRate(), questionReview.getComment());
    }
}

File: qc-domain/qc-review/qc-review-api/src/main/kotlin/com/eager/questioncloud/review/dto/MyQuestionReview.kt
package com.eager.questioncloud.review.dto

import com.eager.questioncloud.review.domain.QuestionReview

class MyQuestionReview(
    val id: Long,
    val rate: Int,
    val comment: String,
) {
    companion object {
        fun from(questionReview: QuestionReview): MyQuestionReview {
            return MyQuestionReview(questionReview.id, questionReview.rate, questionReview.comment)
        }
    }
}
